### PR TITLE
feat: Profile Card

### PR DESCRIPTION
**Pull request by syntag**
Adds `ProfileCard` component and `NeynarProfileCard` wrapper, which uses the Neynar API to fetch user data with `fid` and `viewerFid`.

**New features:**
- Added global theming support for :sunny: light and :crescent_moon: dark modes
- Added `Box`, `VBox`, and `HBox` shared component for layouts

**Demo:**

https://github.com/neynarxyz/react/assets/10079423/05d32ff0-a100-4a93-8488-6f88d10fc4a8

### CODE REVIEW

File: src/components/stories/NeynarProfileCard/NeynarProfileCard.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+export const Primary = Template.bind({});
+Primary.args = {
+  username: "rish",
+  displayName: "rish",
+  avatarImgUrl: "https://i.imgur.com/naZWL9n.gif",
+  bio: "building /neynar 🪐 | neynar.com | /rish",
+  followers: 127364,
+  following: 676,
+  location: "Boston, USA",
```
Comment (manan19) [inline src/components/stories/NeynarProfileCard/NeynarProfileCard.stories.tsx:26]: location isn't available on the protocol, so we should probably remove this

**Review by MrKevinOConnell — COMMENTED** [review]
I think this looks pretty good:

light + dark modes are good - would be good imo to add custom css styling but dont need that for v1
https://github.com/pugson/react-farcaster-embed

File: src/contexts/NeynarContextProvider.tsx [hunk 2]
```diff
@@ -62,20 +61,6 @@ export const NeynarContextProvider: React.FC<NeynarContextProviderProps> = ({
     );
   };
```
Comment (Shreyaschorge) [inline src/contexts/NeynarContextProvider.tsx:63]: Removal of the following code is a breaking change in the SDK.

It'll not add `.theme-light` or `.dark-theme` class to the root thus the theme properties associated with these themes (i.e. `--palette-background`, `--palette-border` etc) will not be present in the DOM.

`withThemeByClassName` will only work for storybook but we need these properties to be present in the DOM as well.

Is there a better way to add these classes to the root of the DOM?
```
useEffect(() => {
    const root = document.querySelector(":root");
    if (root) {
      if (theme === "light") {
        root.classList.add("theme-light");
        // root.classList.remove("theme-dark");
      }
      // else {
      //   root.classList.add("theme-dark");
      //   root.classList.remove("theme-light");
      // }
    }
  }, [theme]);
  ```

File: src/components/stories/NeynarProfileCard.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  onCast: () => {},
+};
+Primary.argTypes = {
+  fid: { table: { disable: true } },
+  viewerFid: { table: { disable: true } },
+  onCast: { table: { disable: true } },
+};
+
+export const WithUser = TemplateWithUser.bind({});
+WithUser.args = {
```
Comment (Shreyaschorge) [inline src/components/stories/NeynarProfileCard.stories.tsx:38]: WithUser is breaking right now, even after being logged in, can we use the user from the  Neynar Context, i.e. if we signed into using NeynarAuthButon that user should be filled into this

@manan19 Should the components in SDK remember the state of the logged-in user? Or do we want to give flexibility to the devs to add their own state or should we do both?

image

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]

File: src/components/stories/NeynarProfileCard.stories.tsx (hunk 3 above)
Comment (Shreyaschorge) [reply src/components/stories/NeynarProfileCard.stories.tsx:38]: This was caused due to `NEYNAR_API_URL` was the part of .env which should not be the case

**Pull request merged**